src/appleseed-max-impl/appleseedoslplugin/osltexture.cpp

```diff
@@ -419,137 +413,38 @@ UVGen* OSLTexture::GetTheUVGen()
-                }
-                break;
-
-              case MaxParam::Color:
-                {
-                    const auto param_value = GetParamBlock(0)->GetColor(max_param.m_max_param_id, t);
-                    params.insert(max_param.m_osl_param_name.c_str(), fmt_osl_expr(to_color3f(param_value)));
-                }
-                break;
-
-              case MaxParam::VectorParam:
-                {
-                    const Point3 param_value = GetParamBlock(0)->GetPoint3(max_param.m_max_param_id, t);
-                    params.insert(max_param.m_osl_param_name.c_str(), fmt_osl_expr(to_vector3f(param_value)));
-                }
-                break;
-
-              case MaxParam::StringPopup:
-                {
-                    std::vector<std::string> fields;
-                    asf::tokenize(param_info.m_options, "|", fields);
-
-                    const int param_value = GetParamBlock(0)->GetInt(max_param.m_max_param_id, t);
-                    params.insert(max_param.m_osl_param_name.c_str(), fmt_osl_expr(fields[param_value]));
-                }
-                break;
-
-              case MaxParam::String:
-                {
-                    const wchar_t* str_value;
-                    GetParamBlock(0)->GetValue(max_param.m_max_param_id, t, str_value, FOREVER);
-                    if (str_value != nullptr)
-                        params.insert(max_param.m_osl_param_name.c_str(), fmt_osl_expr(wide_to_utf8(str_value)));
-                }
-                break;
-            }
-        }
-    }
-
     if (m_has_uv_coords)
     {
         auto uv_transform_layer_name = asf::format("{0}_{1}_uv_transform", material_node_name, material_input_name);
         shader_group.add_shader("shader", "as_max_uv_transform", uv_transform_layer_name.c_str(), get_uv_params(this));
 
+        auto* uv_coord_input = m_shader_info->find_maya_attribute("uvCoord");
+        auto* uv_filter_input = m_shader_info->find_maya_attribute("uvFilterSize");
+
         shader_group.add_connection(
             uv_transform_layer_name.c_str(), "out_outUV",
-            layer_name.c_str(), "in_uvCoord");
+            layer_name.c_str(), uv_coord_input->m_param_name.c_str());
 
         shader_group.add_connection(
             uv_transform_layer_name.c_str(), "out_outUvFilterSize",
-            layer_name.c_str(), "in_uvFilterSize");
+            layer_name.c_str(), uv_filter_input->m_param_name.c_str());
     }
 
-    shader_group.add_shader("shader", m_shader_info->m_shader_name.c_str(), layer_name.c_str(), params);
-    int output_slot_index = GetParamBlock(0)->GetInt(m_shader_info->m_output_param.m_max_param_id, t);
-    m_shader_info->m_output_params[output_slot_index].m_param_name;
-    
+    create_osl_shader(
+        nullptr,
+        shader_group,
+        layer_name.c_str(),
+        m_pblock,
+        m_shader_info);
+
     //TODO: Choose right type of output based on expected input type.
```

Comment:
Insert space after `//`.

Reply:
I'll remove it since we decided to use output selector map